examples/prefect/training.py

```diff
@@ -0,0 +1,51 @@
+import numpy as np
+from prefect import flow, task
+from sklearn import base
+from sklearn.datasets import make_regression
+from sklearn.linear_model import LinearRegression
+from sklearn.model_selection import train_test_split
+
+
+@task
+def make_regression_data(
+    random_state: int, n_samples: int = 100, n_features: int = 1, noise: float = 0.2
+) -> tuple[np.ndarray, np.ndarray]:
+    X, y = make_regression(
+        n_samples=n_samples, n_features=n_features, noise=noise, random_state=random_state
+    )
+    return X, y
+
+
+@task
+def make_train_test_split(
+    X: np.ndarray, y: np.ndarray, random_state: int, test_size: float = 0.2
+) -> tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
+    X_train, X_test, y_train, y_test = train_test_split(
+        X, y, test_size=test_size, random_state=random_state
+    )
+    return X_train, y_train, X_test, y_test
+
+
+@flow
+def prepare_regression_data(
+    random_state: int = 42, n_samples: int = 100, n_features: int = 1, noise: float = 0
+) -> tuple[np.ndarray, ...]:
+    X, y = make_regression_data(random_state=random_state, n_features=n_features, noise=noise)
+    X_train, y_train, X_test, y_test = make_train_test_split(X=X, y=y, random_state=random_state)
+    return X_train, y_train, X_test, y_test
+
+
+@flow
+def prepare_regressor_and_test_data(
+    random_state: int = 42,
+) -> tuple[base.BaseEstimator, np.ndarray, np.ndarray]:
+    X_train, y_train, X_test, y_test = prepare_regression_data(random_state=random_state)
+    model = train_linear_regression(X=X_train, y=y_train)
+    return model, X_test, y_test
+
+
+@task
+def train_linear_regression(X: np.ndarray, y: np.ndarray) -> base.BaseEstimator:
+    model = LinearRegression()
+    model.fit(X, y)
+    return model
```

Comment:
Another candidate for inlining. (Also please define all tasks after the flows, seems to be a convention for Python orchestrators in general.)

Reply:
The thing is if we inline everything the example may become too small to show the integration into Prefect.

And you mean flows after tasks, right? At least thats what the Prefect docs do.

Reply:
Yes, flows after tasks, sorry.